src/test/java/bio/terra/service/resourcemanagement/OneProjectPerProfileIdSelectorTest.java

```diff
@@ -92,9 +97,25 @@ public void shouldGetCorrectIdForBucket() throws Exception {
         String datasetName = "adataset";
         BillingProfileModel billingProfile = ProfileFixtures.randomBillingProfile();
         String projectId = oneProjectPerProfileIdSelector.bucketForFile(datasetName, billingProfile);
-        String expectedProfileId = resourceConfiguration.getProjectId()
+        String expectedProfileId = resourceConfiguration.getDataProjectPrefixToUse()
             + "-" + billingProfile.getProfileName()
             + "-bucket";
         assertThat("Project ID is what we expect", projectId, equalTo(expectedProfileId));
     }
+
+    @Test
+    public void shouldGetCorrectIdForDatasetWithPrefix() throws Exception {
+        String datasetName = "adataset";
+        BillingProfileModel billingProfile = ProfileFixtures.randomBillingProfile();
+        String projectId = oneProjectPerProfileIdSelector.projectIdForDataset(datasetName, billingProfile);
+        String expectedProfileId =
+            resourceConfiguration.getProjectId() + "-" + billingProfile.getProfileName();
+        assertThat("Project ID is what we expect before changing prefix", projectId, equalTo(expectedProfileId));
+        resourceConfiguration.setDataProjectPrefix("PREFIX");
+        String projectIdWithPrefix = oneProjectPerProfileIdSelector.projectIdForDataset(datasetName, billingProfile);
+        String expectedProfileIdWithPrefix =
+            resourceConfiguration.getProjectId() + "-" + billingProfile.getProfileName();
```

Comment:
Would we also want to test the case where the profileId contains the `dataProjectPrefix`?

Reply:
Ah wait, good catch...I think it's maybe testing the wrong thing here...

Reply:
With the logic change it looks like this should use `getDataProjectPrefix` now

Reply:
Sigh, you're totally right.  I had the change locally and somehow it didn't make it into the commit.  Sorry about that

Reply:
No worries! I've done that before as well 😅